src/nvim/statusline.c

```diff
@@ -1678,14 +1678,16 @@ int build_stl_str_hl(win_T *wp, char *out, size_t outlen, char *fmt, char *opt_n
           stl_items[curitem].minwid = -(sattr ? stcp->sign_cul_id ? stcp->sign_cul_id
                                         : sattr->hl_id : (stcp->use_cul ? HLF_CLS : HLF_SC) + 1);
         }
+        size_t buflen = strlen(buf_tmp);
         stl_items[curitem].type = Highlight;
-        stl_items[curitem].start = out_p + strlen(buf_tmp);
+        stl_items[curitem].start = out_p + buflen;
         curitem++;
         if (i == width) {
           str = buf_tmp;
           break;
         }
-        STRCAT(buf_tmp, p);
+        int rc = snprintf(buf_tmp + buflen, sizeof(buf_tmp) - buflen, "%s", p);
+        assert(rc > 0);
```

Comment:
Does this not also need `(void)rc;`?

Reply:
Added, thanks.

Reply:
We can (and should IMO) use `xstrlcpy()` here rather than `snprintf()`.